packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -973,25 +973,24 @@ export class AttributesController extends ConsumptionBaseController {
         return false;
     }
 
-    // TODO: rename this and the respective runtime use case
-    public async getSharedVersionsOfRepositoryAttribute(id: CoreId, peers?: CoreAddress[], onlyLatestVersions = true): Promise<LocalAttribute[]> {
-        let repositoryAttribute = await this.getLocalAttribute(id);
-        if (typeof repositoryAttribute === "undefined") {
+    public async getSharedVersionsOfAttribute(id: CoreId, peers?: CoreAddress[], onlyLatestVersions = true): Promise<LocalAttribute[]> {
+        let sourceAttribute = await this.getLocalAttribute(id);
+        if (typeof sourceAttribute === "undefined") {
```

Comment:
don't we have to throw if the attribute is already shared? before we checked that it's a repository attribute

additionally: does it make sense to call this function for RelationshipAttributes? In that case it would only return third party attributes, right?

Reply:
> don't we have to throw if the attribute is already shared? before we checked that it's a repository attribute
Now we simply return an empty list, if there are no shared versions of the Attribute (in accordance to e.g. getAttributes)

> additionally: does it make sense to call this function for RelationshipAttributes? In that case it would only return third party attributes, right?
This is exactly the desired behavior. The need for this arouse in the parent branch (e.g. https://github.com/nmshd/runtime/pull/76/files#diff-e79d6068574bbea96206a547db0e7d1a54f4b45b3efc1cd453e8d0fb5ff31d59R91), where we need to check if a successor of a RelationshipAttribute was already shared with a third party, before we may share a predecessor as Response to a ReadAttributeRequestItem. But we'll also make use of it within the Attribute deletion, where we need to get all ThirdPartyRelationshipAttributes of a RelationshipAttribute to unset their `shareInfo.sourceAttribute` field. (This is already implemented in the AttributesController, but uses the old name getSharedVersionsOfRepositoryAttribute to receive possible predecessors. So if I am not mistaken, we should currently have a bug here: In case of a succeeded ThirdPartyRelationshipAttribute, an error should be thrown trying to delete the predecessor ThirdPartyRelationshipAttribute. But I think before this parent branch is merged, it isn't even possible to succeed ThirdPartyRelationshipAttributes, so this case hasn't occured so far.)